pkg/util/ip_reservation.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+)
+
+// GetUnreservedIPBlock returns an unreserved /29 IP block. It accepts the list of currently reserved
+// IPs and the requested CIDR as arguments and returns the /29 IP available in that CIDR
+// Potential error cases: 1) No /29 Block in the CIDR is unreserved
+// Parsing the CIDR resulted in an error
+func GetUnreservedIPBlock(reservedIPRanges map[string]bool, cidr string) (string, error) {
+	ip, ipnet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return "", fmt.Errorf("Unable to parse CIDR %s", cidr)
+	}
+	buffer := bytes.NewBufferString("")
+	for cidrIPBlock := cloneIP(ip.Mask(ipnet.Mask)); ipnet.Contains(cidrIPBlock) && err == nil; err = incrementIP(cidrIPBlock, 8) {
```

Comment:
What is the number `8` in this case? Maybe we should make a const for it or a comment here to specify what it is. I noticed in the function definition its of type `byte`, this is somewhat unusual and is worth clarifying so there are no gotchas in the future

Reply:
8 is used to get to the next /29 address. I will add a constant for this.